blocks/bricks/cost.py

```diff
@@ -18,6 +18,19 @@ def apply(self, y, y_hat):
 
 
 @add_metaclass(ABCMeta)
+class Softmax(Brick):
+    @application
+    def apply(self, x):
+        return tensor.nnet.softmax(x)
+
+    @application
+    def categorical_cross_entropy(self, x, y_hat):
```

Comment:
The input is `y` I think (`y_hat` would be the output of the softmax, while `y` is the actual distribution i.e. the labels)